python3/smb/utils/md4.py

```diff
@@ -243,14 +243,6 @@ def f1(a, b, c, d, k, s, X): return ROL(a + F(b, c, d) + X[k], s)
 def f2(a, b, c, d, k, s, X): return ROL(a + G(b, c, d) + X[k] + U32(0x5a827999), s)
 def f3(a, b, c, d, k, s, X): return ROL(a + H(b, c, d) + X[k] + U32(0x6ed9eba1), s)
 
-#--------------------------------------------------------------------
-# helper function
-def int_array2str(array):
-        str = ''
-        for i in array:
-            str = str + chr(i)
```

Comment:
the root problem that we were seeing is that chr(i) was blowing up, because this depends on `U32.__int__()`, which seems to not be supported anymore.  An alternate fix might be to add a non-dunder as_int() method to U32, or  (looking at what `U32.__int__()` actually does) to use int(i.truth()).

Reply:
(just to follow up, FWIW, this seems to be the breaking change, in the 3.10 change docs: "Builtin and extension functions that take integer arguments no longer accept Decimals, Fractions and other objects that can be converted to integers only with a loss (e.g. that have the __int__() method but do not have the __index__() method). (Contributed by Serhiy Storchaka in bpo-37999.)")